### DISCUSSION TITLE
The container stops working after being updated and started.

### DISCUSSION THREAD

**Discussion by milesxia** [McCloudS/subgen#153] [General]
Hi, I'm back. Today, I found that my Subgen has stopped working. After pulling a new Docker image and starting the container, the logs showed this content before it stopped running. What should I do next?

==========
== CUDA ==
==========

CUDA Version 12.3.2

Container image Copyright (c) 2016-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.

This container image and its contents are governed by the NVIDIA Deep Learning Container License.
By pulling and using the container, you accept the terms and conditions of this license:
https://developer.nvidia.com/ngc/nvidia-deep-learning-container-license

A copy of this license is made available in this container at /NGC-DL-CONTAINER-LICENSE for your convenience.

WARNING: The NVIDIA Driver was not detected.  GPU functionality will not be available.
   Use the NVIDIA Container Toolkit to start this container with GPU support; see
   https://docs.nvidia.com/datacenter/cloud-native/ .

subgen.env file not found. Please run prompt_and_save_env_variables() first.
Downloading subgen.py from GitHub branch main...
File downloaded successfully to subgen.py
File downloaded successfully to language_code.py
Launching subgen.py
INFO:root:Subgen v2024.12.6
INFO:root:Starting Subgen with listening webhooks!
INFO:root:Transcriptions are limited to running 2 at a time
INFO:root:Running 4 threads per transcription
INFO:root:Using cpu to encode
INFO:root:Using faster-whisper
INFO:root:Starting to search folders to see if we need to create subtitles.
ERROR:root:An error occurred while checking the file with pyav: 
ERROR:root:An error occurred while checking the file with pyav: 
Traceback (most recent call last):
  File "/subgen/subgen.py", line 1337, in has_audio
    with av.open(file_path) as container:
  File "av/container/core.pyx", line 369, in av.container.core.open
  File "av/container/core.pyx", line 236, in av.container.core.Container.__cinit__
  File "av/container/core.pyx", line 256, in av.container.core.Container.err_check
  File "av/error.pyx", line 428, in av.error.err_check
av.error.InvalidDataError: [Errno 1094995529] Invalid data found when processing input: '/media/link/cartoon/动漫/蜡笔小新 合集 (1992)/Season 12/蜡笔小新 合集 - S12E23 - 第23集.mkv'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/subgen/subgen.py", line 1431, in <module>
    transcribe_existing(transcribe_folders)
  File "/subgen/subgen.py", line 1405, in transcribe_existing
    gen_subtitles_queue(path_mapping(file_path), transcribe_or_translate, forceLanguage)
  File "/subgen/subgen.py", line 973, in gen_subtitles_queue
    if not has_audio(file_path):
  File "/subgen/subgen.py", line 1348, in has_audio
    except (av.AVError, UnicodeDecodeError):
AttributeError: module 'av' has no attribute 'AVError'. Did you mean: 'OSError'?
Traceback (most recent call last):
  File "/subgen/launcher.py", line 171, in <module>
    main()
  File "/subgen/launcher.py", line 166, in main
    subprocess.run([f'{python_cmd}', '-u', 'subgen.py'], check=True)
  File "/usr/lib/python3.10/subprocess.py", line 526, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['python3', '-u', 'subgen.py']' returned non-zero exit status 1.

**Comment by McCloudS** [comment]
Good news, I fixed the error.  PyAV recently had an update removing av.AVError.  Bad news, you’ll probably still get an error on that file.  But, the new error catch can give us some insight as to why.  Could be a corrupted file, bad codec, etc…

**Reply by MediaMoots** [reply]
hello, would it be possible to push a new docker build with this fix?

**Reply by McCloudS** [reply]
> hello, would it be possible to push a new docker build with this fix?

Building now.

**Reply by milesxia** [reply]
Alright, I will try updating the image right now. Thank you for resolving the issue. If I encounter any specific error messages, I’ll reach out to you again to discuss what might be causing the problem and how to fix it.

**Reply by milesxia** [reply]
INFO:root:File: /media/movie/Out_of_the_blue.1991.mp4 was added
ERROR:root:An error occurred while checking the file with pyav: 
ERROR:root:An error occurred while checking the file with pyav: 
ERROR:root:An error occurred while checking the file with pyav: 
INFO:root:task_queue.put(task)(/media/movie/Out_of_the_blue.1991.mp4, transcribe, Chinese)
INFO:root:Task /media/movie/Out_of_the_blue.1991.mp4 is being handled by Subgen.
INFO:root:Added Out_of_the_blue.1991.mp4 for transcription.
INFO:root:Transcribing file: Out_of_the_blue.1991.mp4
INFO:root:Transcribing file language: Chinese
INFO:root:Error processing or transcribing /media/movie/Out_of_the_blue.1991.mp4 in Chinese: Invalid compute type: cpu



Subgen is running normally now, but there are still errors in the logs. I tried adding a new video, but the transcription process still hasn't started.

**Reply by McCloudS** [reply]
Your `COMPUTE_TYPE` is invalid, you're better off setting it to `auto`.

**Reply by milesxia** [reply]
Since I'm using a QNAP NAS device, it doesn't have a GPU for processing, so I forced it to use the CPU instead.How amazing, it works properly after setting it to auto!
